runtime/restart/monitor/monitor.go

```diff
@@ -232,10 +232,10 @@ func (m *monitor) monitor(ctx context.Context) ([]change, error) {
 			}
 		}
 
-		// Task or Status return error, only desired to stop
+		// Task or Status return error, only desired to running
 		if err != nil {
 			logrus.WithError(err).Error("monitor")
-			if desiredStatus != containerd.Stopped {
+			if desiredStatus == containerd.Stopped {
```

Comment:
if desiredStatus is running, the `"on-failure"` case will get empty status in the Line 245.
I am not sure that there is a way to store the status after delete. Anyway, it is safe to use with nerdctl.
Maybe we can add the comment about known issue.

Reply:
An empty status has no effect on `on-failure`, which expects a status whose `ExitStatus` is not 0.

Reply:
I get it, I added the comment about known issue in 243-245 .